### PR TITLE
feat: add profiling options

### PR DESCRIPTION
**Pull request by Rjected**
Adds a subset of the profiling options. This requires the `profiling` feature to be enabled on `tikv-jemalloc-ctl` to be used.

Also adds the new `macos-14` runner for the `aarch64-apple-darwin` target to CI.

### CODE REVIEW

**Review by BusyJay — COMMENTED** [review]
Why only read access is needed?

**Comment by Rjected** [comment]
> Why only read access is needed?

Jemalloc docs say the following:

```
opt.prof (bool) r-
opt.lg_prof_interval (ssize_t) r-
opt.lg_prof_sample (size_t) r-
opt.prof_final (bool) r-
opt.prof_leak (bool) r-
```

Write access would be nice, but it doesn't look like we can have write access for these `opt`s

File: jemalloc-ctl/src/profiling.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
+//! `jemalloc`'s run-time configuration for profiling-specific settings.
+//!
+//! These settings are controlled by the `MALLOC_CONF` environment variable.
+#![cfg(feature = "profiling")]
```
[the remark this answers is not in the document]
Comment (Rjected) [reply jemalloc-ctl/src/profiling.rs]: removed

**Review by BusyJay — APPROVED** [review]

**Comment by BusyJay** [comment]
Thanks!

**Pull request merged**